### PR TITLE
Add AniDex torrent provider

### PR DESCRIPTION
**Pull request by medariox**
- [x] PR is based on the DEVELOP branch
- [x] Don't send big changes all at once. Split up big PRs into multiple smaller PRs that are easier to manage and review
- [x] Read the [contribution guide](https://github.com/pymedusa/Medusa/blob/master/.github/CONTRIBUTING.md)

### CODE REVIEW

**Comment by medariox** [comment]
To be merged after: https://github.com/pymedusa/Medusa/pull/2693

File: medusa/providers/torrent/html/anidex.py [hunk 1]
```diff
@@ -145,13 +149,13 @@ def parse(self, data, mode):
...
-                        logger.log('Found result: {0} with {1} seeders and {2} leechers'.format
-                                   (title, seeders, leechers), logger.DEBUG)
+                        log.debug('Found result: {0} with {1} seeders and {2} leechers',
+                                  title, seeders, leechers)
 
                     items.append(item)
                 except (AttributeError, TypeError, KeyError, ValueError, IndexError):
-                    logger.log('Failed parsing provider. Traceback: {0!r}'.format
-                               (traceback.format_exc()), logger.ERROR)
+                    log.error('Failed parsing provider. Traceback: {0!r}',
```
Comment (p0psicles) [inline medusa/providers/torrent/html/anidex.py]: I prefer to use log.exception() in these situations. @labrys what you think?

Comment (labrys) [inline medusa/providers/torrent/html/anidex.py]: Agreed.  Its much cleaner to do `log.exception('Failed parsing provider')`.

Comment (medariox) [inline medusa/providers/torrent/html/anidex.py]: It's the same for me, but we should decide what to use. I saw @labrys still replacing it with log.error?

Comment (labrys) [inline medusa/providers/torrent/html/anidex.py]: For my replaces its mostly just mass replaces with regex to deprecate the old brace message format so we can remove it.  Not spending much effort on content of messages.

**Pull request merged**